x-pack/platform/plugins/shared/fleet/server/services/cloud_connector.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/core/server';
+import type { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+
+import type {
+  CloudConnectorSO,
+  CloudConnectorListOptions,
+  CloudProvider,
+  CloudConnectorVars,
+  CloudConnectorSecretVar,
+} from '../../common/types/models/cloud_connector';
+import type { CloudConnectorSOAttributes } from '../types/so_attributes';
+import type { CreateCloudConnectorRequest } from '../routes/cloud_connector/handlers';
+import { CLOUD_CONNECTOR_SAVED_OBJECT_TYPE } from '../../common/constants';
+
+import { appContextService } from './app_context';
+
+export interface CloudConnectorServiceInterface {
+  create(
+    soClient: SavedObjectsClientContract,
+    cloudConnector: CreateCloudConnectorRequest
+  ): Promise<CloudConnectorSO>;
+  getList(
+    soClient: SavedObjectsClientContract,
+    options?: CloudConnectorListOptions
+  ): Promise<CloudConnectorSO[]>;
+}
+
+export class CloudConnectorService implements CloudConnectorServiceInterface {
+  private static readonly EXTERNAL_ID_REGEX = /^[a-zA-Z0-9]{20}$/;
+
+  protected getLogger(...childContextPaths: string[]): Logger {
+    return appContextService
+      .getLogger()
+      .get('[Cloud Connector API] CloudConnectorService', ...childContextPaths);
+  }
+
+  async create(
+    soClient: SavedObjectsClientContract,
+    cloudConnector: CreateCloudConnectorRequest
+  ): Promise<CloudConnectorSO> {
+    const logger = this.getLogger('create');
+    logger.debug('Creating cloud connector');
+
+    try {
+      logger.info('Creating cloud connector');
+
+      // Extract cloud variables from package policy
+      const { cloudProvider, vars, name } = this.extractCloudVars(cloudConnector);
+
+      if (!vars || Object.keys(vars).length === 0) {
+        logger.error(`Package policy must contain ${cloudProvider} input vars`);
+        throw new Error(
```

Comment:
best to create a new error class here and throw that instead:
```
x-pack/platform/plugins/shared/fleet/server/errors
```

that goes for everywhere in this PR where a cloud connector error is being throw